internal/grader/reject.go

```diff
@@ -45,10 +45,48 @@ func (this *RejectWindowMax) fullString(now timestamp.Timestamp) string {
 		nextTime.SafeMessage(), deltaString)
 }
 
-func checkForRejection(assignment *model.Assignment, submissionPath string, user string, message string) (RejectReason, error) {
+type RejectLateWithoutAllow struct {
+	AssignmentName string
+	DueDate        timestamp.Timestamp
+}
+
+func (this *RejectLateWithoutAllow) String() string {
+	return this.fullString(timestamp.Now())
+}
+
+func (this *RejectLateWithoutAllow) fullString(now timestamp.Timestamp) string {
+	deltaMS := now.ToMSecs() - this.DueDate.ToMSecs()
+	deltaString := time.Duration(deltaMS * int64(time.Millisecond)).String()
+
+	return fmt.Sprintf("Attempting to submit assignment (%s) late without 'allow late'."+
+		" It was due on %s (which was %s ago). Include the 'allow late' flag to submit an assignment late."+
+		" See your interface's help section for more information.",
+		this.AssignmentName, this.DueDate.SafeMessage(), deltaString)
+}
+
+func checkForRejection(assignment *model.Assignment, submissionPath string, user string, message string, allowLate bool) (RejectReason, error) {
+	reason := checkLateSubmission(assignment, allowLate)
+	if reason != nil {
+		return reason, nil
+	}
+
 	return checkSubmissionLimit(assignment, user)
 }
 
+func checkLateSubmission(assignment *model.Assignment, allowLate bool) RejectReason {
+	now := timestamp.Now()
```

Comment:
Do the nil check first.
(It's small, but generally we want to see if we have to do work before we do it.)